sunpy/map/sources/hinode.py

```diff
@@ -60,6 +60,20 @@ def __init__(self, data, header, **kwargs):
         self.plot_settings['norm'] = ImageNormalize(
             stretch=source_stretch(self.meta, LogStretch()), clip=False)
 
+    @property
+    def coordinate_system(self):
+        """
+        Override the default implementation to handle SOTMap-specific logic for CTYPE values.
+        """
+        if self.meta:
```

Comment:
What is this for?

Reply:
https://github.com/sunpy/sunpy/pull/7955#discussion_r1899842589
For this comment

Reply:
If self.meta is None the coordinate_system method wont be overriden

Reply:
Ah I misspoke. It might be empty but never none.

I mean to say what if `self.meta.get('ctype1', None)` is None. How does the line below it work?

Reply:
> I mean to say what if `self.meta.get('ctype1', None)` is None. How does the line below it work?
Ah ic. The way it was handled in `mapbase.py` was 

```python
ctype1 = self.meta.get('ctype1', None)
if not ctype1:
    warn_metadata("Missing CTYPE1 from metadata, assuming CTYPE1 is HPLN-TAN")
    ctype1 = 'HPLN-TAN'

ctype2 = self.meta.get('ctype2', None)
if not ctype2:
    warn_metadata("Missing CTYPE2 from metadata, assuming CTYPE2 is HPLT-TAN")
    ctype2 = 'HPLT-TAN'
```


But fromhttps://github.com/sunpy/sunpy/pull/7955#discussion_r1899732479
I thought we should keep that logic only in mapbase and warnings(warn_metadata) shouldnt be added in the map sources

Reply:
This logic is different and should stay in mapbase.